src/Umbraco.Web/Models/Mapping/ContentTypeMapDefinition.cs

```diff
@@ -223,7 +223,11 @@ private static void Map(PropertyTypeBasic source, PropertyType target, MapperCon
             target.DataTypeKey = source.DataTypeKey;
             target.Mandatory = source.Validation.Mandatory;
             target.ValidationRegExp = source.Validation.Pattern;
-            target.Variations = source.AllowCultureVariant ? ContentVariation.Culture : ContentVariation.Nothing;
+            target.Variations = source.AllowCultureVariant
+                // Set culture flag
+                ? target.Variations | ContentVariation.Culture
+                // Unset culture flag
+                : target.Variations & ~ContentVariation.Culture;
```

Comment:
I know this looks beautiful but it is extremely unreadable and I personally am not familiar with this notation, so I have no idea what is going on. For readability can we please turn this into multiple `if`s and whatever the `~` means? :-)

Reply:
Thanks for the review. The original code also used a ternary `if`-statement which is why I kept it that way. But will change soon then.

Regarding the `~` operator; this is the [bitwise complement operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/bitwise-and-shift-operators#bitwise-complement-operator-), so it inverts all the bits (e.g. `1100` would become `0011`). So by performing an AND (`&`) with the complement of `Culture` we remove the `Culture` flag.

Using bitwise operations is the normal way of manipulating flags enum types, like `target.Variations`. This is also explained by [Microsoft here](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/enumeration-types#enumeration-types-as-bit-flags). 

I would advice against removing these operators just because bitwise operators are not very well known. 

With respect to readability it could however be nice to add an extension method like `UnsetFlag()` in the [`EnumExtensions`](https://github.com/umbraco/Umbraco-CMS/blob/v8/dev/src/Umbraco.Core/EnumExtensions.cs) that you guys already have.

Reply:
Thanks @PerplexDaniel! 
Cool, thanks for the insight!

I agree, that the `bitwise complement operator` is absolutely fine then, make sure to leave a code comment. 

And yes, I really like the idea of adding an extension method, with a description of what it does and why.

I love ternary operators, but it gets a bit unreadable when they seemingly start to do more than one thing in the if/else clauses. Clearly my lack of knowledge on bitwise operators led me to believe there was multiple nested if/else statements here. So they're still fine to use! 

If you could refactor this into an extension method with some method documentation then this should be good to go! 👍